### DISCUSSION TITLE
Please tell me how to change my working hours chart on the dashboard to the working hours chart for all users.

### DISCUSSION THREAD

**Discussion by seleondevel** [kimai/kimai#5116] [Plugins]
![화면 캡처 2024-10-15 150507](https://github.com/user-attachments/assets/da633cbe-c165-4f5e-aa29-1b396f178ee1)

Hello.

There is a chart showing my working hours on the Kimai dashboard.
I want to change this chart to show the working hours of the entire user, how can I work on it? Any advice?

**Comment by kevinpapst** [comment]
Start by copying the widget https://github.com/kimai/kimai/blob/main/src/Widget/Type/PaginatedWorkingTimeChart.php

**Reply by seleondevel** [reply]
I created a widget type in the plug-in, and I also created a template, but I get a template path error for the Unable to find template "widget/widget-allworkingtimecart.html.twig". Do I need to create a WidgetExtension, too?

**Reply by kevinpapst** [reply]
You have to correctly use your template, same as in a controller. Prefix it with your bundle:
https://github.com/kimai/DemoBundle/blob/main/Widget/DemoWidget.php#L70

**Reply by seleondevel** [reply]
![화면 캡처 2024-10-16 152422](https://github.com/user-attachments/assets/1d206865-9ff8-4b30-a217-170ec159a260)

I've added a chart of working hours, but when I hit the left and right buttons, I get a loading error on the selection page. Is there anything else I need to work on?

**Reply by seleondevel** [reply]
I've changed the left and right button paths, but I'm having trouble with the existing work hour widget disappearing.

**Reply by seleondevel** [reply]
CRITICAL: Uncaught PHP Exception Twig\Error\RuntimeError: "Variable "options" does not exist."
I can't solve this problem. Should I copy DailyWorkingTimeChart.php as well?

**Reply by seleondevel** [reply]
I solved it without much difficulty, thank you.
